crates/sui-types/src/rpc_proto_conversions.rs

```diff
@@ -1725,13 +1725,49 @@ impl From<crate::transaction::GenesisObject> for Object {
 // ObjectReference
 //
 
-fn object_ref_to_proto(value: crate::base_types::ObjectRef) -> ObjectReference {
-    let (object_id, version, digest) = value;
-    let mut message = ObjectReference::default();
-    message.object_id = Some(object_id.to_canonical_string(true));
-    message.version = Some(version.value());
-    message.digest = Some(digest.to_string());
-    message
+pub trait ObjectRefExt {
+    fn to_proto(self) -> ObjectReference;
+}
+
+pub trait ObjectReferenceExt {
+    fn try_to_object_ref(&self) -> Result<crate::base_types::ObjectRef, anyhow::Error>;
+}
+
+impl ObjectRefExt for crate::base_types::ObjectRef {
+    fn to_proto(self) -> ObjectReference {
+        let (object_id, version, digest) = self;
+        let mut message = ObjectReference::default();
+        message.object_id = Some(object_id.to_canonical_string(true));
+        message.version = Some(version.value());
+        message.digest = Some(digest.to_string());
+        message
+    }
+}
+
+impl ObjectReferenceExt for ObjectReference {
+    fn try_to_object_ref(&self) -> Result<crate::base_types::ObjectRef, anyhow::Error> {
+        use anyhow::Context;
+
+        let object_id = self
+            .object_id_opt()
+            .ok_or_else(|| anyhow::anyhow!("missing object_id"))?;
+        let object_id = crate::base_types::ObjectID::from_hex_literal(object_id)
+            .with_context(|| format!("Failed to parse object_id: {}", object_id))?;
+
+        let version = self
+            .version_opt()
+            .ok_or_else(|| anyhow::anyhow!("missing version"))?;
+        let version = crate::base_types::SequenceNumber::from(version);
+
+        let digest = self
+            .digest_opt()
+            .ok_or_else(|| anyhow::anyhow!("missing digest"))?;
+        let digest = digest
+            .parse::<crate::digests::ObjectDigest>()
+            .with_context(|| format!("Failed to parse digest: {}", digest))?;
+
+        Ok((object_id, version, digest))
+    }
```

Comment:
I think there are free functions in this file to do this, maybe worth checking to dedup.